sihl/src/contract_database.ml

```diff
@@ -8,29 +8,81 @@ let name = "database"
 exception Exception of string
 
 module type Sig = sig
-  val prepare_requests
-    :  string
-    -> string
-    -> string
-    -> string
+  (** ['a search_request] is a prepared SQL statement that can be used to sort,
+      filter and paginate (= search) a collection. *)
+  type 'a prepared_search_request
+
+  val prepare_requests : string -> string -> string -> string
+    (* Deprecated in 0.6.0 *)
+    [@@deprecated "Use prepare_search_request instead"]
```

Comment:
@aronerben Let's do all deprecations like this. It won't compile when using it, but the compiler can be used to provide an upgrade path.

Reply:
@jerben Good approach!